examples/CMakeLists.txt

```diff
@@ -0,0 +1,100 @@
+cmake_minimum_required(VERSION 3.1...3.17)
+project(QuicheExamples LANGUAGES C)
+set(CMAKE_MODULE_PATH "${CMAKE_SOURCE_DIR}/cmake/modules")
+
+# Default to a Debug build.
+if(NOT DEFINED CMAKE_BUILD_TYPE)
+  set(CMAKE_BUILD_TYPE Debug CACHE STRING "Choose the type of build" FORCE)
+  set_property(CACHE CMAKE_BUILD_TYPE PROPERTY STRINGS
+    "None" "Debug" "Release" "MinSizeRel" "RelWithDebInfo")
+endif()
+
+if(NOT CMAKE_C_FLAGS AND CMAKE_C_COMPILER_ID MATCHES "GNU|Clang")
+  set(CMAKE_C_FLAGS "-Wall -Werror -pedantic -fsanitize=address")
+endif()
+
+if(MSVC AND CMAKE_GENERATOR_PLATFORM STREQUAL x64)
+  set(CARGO_TARGET x86_64-pc-windows-msvc)
+elseif(MSVC AND CMAKE_GENERATOR_PLATFORM STREQUAL Win32)
+  set(CARGO_TARGET i686-pc-windows-msvc)
+endif()
+
+set(SOURCE_DIR      "${CMAKE_CURRENT_SOURCE_DIR}/../src")
+set(BUILD_DIR       "${CMAKE_CURRENT_BINARY_DIR}/build")
+set(QUICHE_LIBNAME "${CMAKE_STATIC_LIBRARY_PREFIX}quiche${CMAKE_STATIC_LIBRARY_SUFFIX}")
+if(CARGO_TARGET)
+  set(CARGO_OPTS --target=${CARGO_TARGET})
+  set(QUICHE_LIB  "${BUILD_DIR}/${CARGO_TARGET}/debug/${QUICHE_LIBNAME}")
+else()
+  set(QUICHE_LIB  "${BUILD_DIR}/debug/${QUICHE_LIBNAME}")
+endif()
+
+
+set_property(DIRECTORY APPEND PROPERTY ADDITIONAL_CLEAN_FILES "${BUILD_DIR}")
+
+file(GLOB_RECURSE QUICHE_SOURCES "${SOURCE_DIR}/*.rs")
+add_custom_target(quiche-build DEPENDS "${QUICHE_LIB}")
+add_custom_command(OUTPUT "${QUICHE_LIB}"
+  COMMAND cargo build --target-dir "${BUILD_DIR}" ${CARGO_OPTS}
+  DEPENDS ${QUICHE_SOURCES}
+  WORKING_DIRECTORY "${CMAKE_CURRENT_SOURCE_DIR}"
+  USES_TERMINAL
+  VERBATIM)
+
+add_library(quiche STATIC IMPORTED)
+set_target_properties(quiche PROPERTIES IMPORTED_LOCATION "${QUICHE_LIB}")
+add_dependencies(quiche quiche-build)
+target_include_directories(quiche INTERFACE "${CMAKE_CURRENT_SOURCE_DIR}/../include")
+if(APPLE)
+  # For SecRandomCopyBytes used by ring::rand::SystemRandom.
+  set_target_properties(quiche PROPERTIES
+    INTERFACE_LINK_LIBRARIES "-framework Security")
+elseif(WIN32)
+  # Crypt32 for CertOpenSystemStoreA used by src/tls.rs, Userenv for std::env,
+  # Ws2_32 for getaddrinfo, Advapi32 for RtlGenRandom.
+  set_target_properties(quiche PROPERTIES
+    INTERFACE_LINK_LIBRARIES "Crypt32;Userenv;Ws2_32;Advapi32")
+  # cc-rs up to at least 1.0.53 adds /MD which results in a LNK4098 warning
+  # because CMake adds a conflicting /MDd option in debug mode. Ensure msvcrtd
+  # is linked in Debug builds by skipping msvcrt (normally added via /MD).
+  # (target_link_options requires CMake 3.13)
```

Comment:
Does it mean we need to update minimum version at L1?

Reply:
Probably yes. Windows and macOS generally have up-to-date CMake versions, but unfortunately Linux distros are often behind, including Ubuntu as used in CI.

I can increase the minimum version for Windows and leave others unchanged.